proxy/src/bind.rs

```diff
@@ -132,7 +132,7 @@ pub struct RebindTls<B> {
 
 pub type Service<B> = BoundService<B>;
 
-pub type Stack<B> = WatchService<Option<tls::ClientConfig>, RebindTls<B>>;
+pub type Stack<B> = WatchService<Conditional<tls::ClientConfig, tls::ReasonForNoTls>, RebindTls<B>>;
```

Comment:
It seems worth having a named `type ConditionalClientTls = ...` that reduces some of the boilerplate.

Reply:
> It seems worth having a named type ConditionalClientTls = ... that reduces some of the boilerplate.

We did already have `ConditionalConnectionConfig` but the `bind` module can't use it for reasons documented in the commit I just pushed to this PR. I added a `ConditionalTlsClientConfig` for the `bind` submodule's private use.